subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1600,6 +1612,18 @@ static int init_reset(void)
 		lll_adv_data_init(&ll_adv[handle].lll.scan_rsp);
 	}
 
+#if defined(CONFIG_BT_HCI_RAW)
+	/* Make sure that set #0 is initialized with empty legacy PDUs. This is
+	 * especially important if legacy HCI interface is used for advertising
+	 * because it allows to enable advertising without any configuration,
+	 * thus we need to have PDUs already initialized.
+	 *
+	 * Note: not needed if Zephyr host is used since it always configures
```

Comment:
Should we really rely on the Host doing this? Why not do this unconditionally?
cc @joerchan

Reply:
I reverted this back to original version (unconditional init). This does not really do any harm and has no real impact on power consumption but just makes code cleaner. It can be optimized later especially if "trusted host" setting is introduced as suggested before.